src/pyggdrasil/tree_inference/_logprob_validator.py

```diff
@@ -0,0 +1,173 @@
+"""Implementation of the log-probability functions.
+
+The log-probability functions are used to calculate the log-probability of a tree.
+
+Implements a dumb version of the log-probability function, which is used for testing.
+
+This version attempts to do it by intuition without looking ta the paper.
+"""
+import jax
+import jax.numpy as jnp
+
+import pyggdrasil.tree_inference._tree as tr
+
+from pyggdrasil.tree_inference import Tree, ErrorRates
```

Comment:
This is a place where a potential circular import could appear. One way around this would be to import these from `_interface` submodule

Reply:
Fixed!